src/main/kotlin/io/github/inductiveautomation/kindling/questdb/QuestDbView.kt

```diff
@@ -0,0 +1,329 @@
+        databaseName = "qdb"
+        user = PG_USER
+        password = PG_PASS
+        portNumbers = intArrayOf(pgPort)
+
+        connection
+    }
+
+    private val pageLabel: JLabel
+
+    @Suppress("SqlResolve")
+    val tables: List<Table> = connection
+        .executeQuery("""SELECT table_name FROM tables();""")
+        .toList { resultSet ->
+            resultSet.getString("table_name")
+        }.mapNotNull { tableName ->
+            try {
+                val columns = connection
+                    .executeQuery("""SHOW COLUMNS FROM "$tableName";""")
+                    .toList { rs ->
+                        Column(
+                            name = rs.getString("column"),
+                            type = rs.getString("type"),
+                            notNull = false,
+                            defaultValue = null,
+                            primaryKey = rs.getBoolean("upsertKey"),
+                            hidden = false,
+                            _parent = { sortableTree.root },
+                        )
+                    }
+                val size: Long
+                connection.executeQuery("""SELECT diskSize FROM table_storage() WHERE tableName = '$tableName'""").use { rs ->
+                    // Move the cursor to the first row. If there's no row, default to 0.
+                    size = if (rs.next()) rs.getLong("diskSize") else 0L
+                }
+                Table(
+                    name = tableName,
+                    _parent = { sortableTree.root },
+                    columns = columns,
+                    size = size,
+                )
+            } catch (e: Exception) {
+                println("Warning: Could not process table '$tableName'. Error: ${e.message}")
+                null
+            }
+        }
+
+    private val sortableTree = SortableTree(tables)
+    private val query = RSyntaxTextArea().apply {
+        syntaxEditingStyle = SyntaxConstants.SYNTAX_STYLE_SQL
+
+        theme = Theme.currentValue
+
+        Theme.addChangeListener { newTheme ->
+            theme = newTheme
+        }
+    }
+
+    private val execute = Action(
+        name = "Execute",
+        description = "Execute (${if (SystemInfo.isMacOS) "⌘" else "Ctrl"} + Enter)",
+        icon = FlatActionIcon("icons/bx-subdirectory-left.svg"),
+        accelerator = KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, menuShortcutKeyMaskEx),
+    ) {
+        results.result = if (!query.text.isNullOrEmpty()) {
+            try {
+                connection.executeQuery(query.text)
+                    .use { resultSet ->
+                        val columnCount = resultSet.metaData.columnCount
+                        val names = List(columnCount) { i ->
+                            resultSet.metaData.getColumnName(i + 1)
+                        }
+                        val types = List(columnCount) { i ->
+                            val isTimestamp = names[i].containsInOrder("tsmp", true)
```

Comment:
Same with this isTimestamp stuff - makes sense for generic sqlite view, probably not needed for quest